Tutorials/Level1Tutorial.md

```diff
@@ -0,0 +1,793 @@
+        $(CXX) $(OPTIMIZE_FLAGS) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $@ *.o
+%.o:%.cc $(FOO_HEADERS)
+        $(CXX) $(OPTIMIZE_FLAGS) $(CXXFLAGS) $(CPPFLAGS) -c $<
+install: libfoo.so
+        sst-register foo foo_LIBDIR=$(CURDIR)
+clean:
+        rm -Rf *.o libfoo.so
+```
+
+## Basic Runtime Functions <a name="BasicRuntimeFunctions"></a>
+
+As we see from the commands above, executing a simulation with SST 
+is as simple as passing a simulation script or JSON file to the `sst` 
+command.  However, in order to do so, we must first develop an SST 
+simulation in Python or JSON format.  In this section, we will provide 
+a basic overview of utilizing Python to create a sample simulation 
+input file.  Additional, advanced Python scripting topics will be discussed 
+in the Level 2 Tutorial.
+
+Each Python-based simulation input script has requires at least three 
+key sections as outlined below:
+* **Python imports**: Each simulation input file must import the `sst` Python
+library.  This provides the fundamental component configuration options 
+and functions that link directly to the SST-Core and any configured elements.
+* **SST Core Options**: The core options section describes the top-level configuration 
+options for the SST-Core.  This includes items such as the SST program options 
+that set the baseline clock candence and simulation runtime.
+* **SST Component Configuration**: The SST component configuration section contains 
+the bulk of the information in a Python simulation script.  This includes the 
+definition, parameters and any required subcomponents for each target component 
+in the simulation.
+
+The following example is a simple SST configuration script that imports 
+the required SST functions, sets two SST core program options and initializes 
+a single SST component.  The first step in creating a new simulation input 
+is to utilize the `import sst` statement.  This ensures that all the required 
+Python functions for SST are enabled.  Users may also utilize additional Python 
+import statements in order to enable additional Python-derived functionality 
+within the simulation script.  In this manner, simulation scripts are **not** 
+limited to SST functionality.  All normal Python syntatic operations may 
+be utilized as with a traditional Python script.  
+
+The next section includes the SST Core Options.  These options drive the 
+top-level simulation configuration that are utilized across components.  Each 
+of the core program options are set with the Python function 
+`sst.setProgramOptions("Option", "Value)`.  The two most commonly 
+used program options are the `timebase` and `stop-at` options.  The 
+`timebase` option sets the baseline global clock cadence by which 
+SST will govern the core simulation components.  Each component has the ability 
+to set higher or lower fidelity clocks governing local event timing, but the 
+global clock drives the top-level simulation cadence.  Further, the `stop-at` 
```

Comment:
This is not quite right. SST is DES, it doesn't have a global clock or any "core components" that are governed by a clock. However, it does have a global timestamp that is used to keep track of the current time (it jumps from time to time according to the events as you'd expect in DES), and setting timebase determines the units of that global timestamp.
Setting a higher fidelity clock than the global timebase will yield a fatal error in SST as the timestamp can't accurately represent any elapsed time interval less than the timebase. 

By default the timebase is 1ps. There are a few uncommon cases in which you'd want to set the parameter manually: 
(1) a component needs a clock with a smaller time period (in which case, set lower)
(2) you need to simulate a time period that is greater than ~2/3 of a year (max value of 1ps in a uint64_t). In this case, increase the timebase but note that any clocks will be constrained to the new timebase at minimum.